### DISCUSSION TITLE
4bb: 4clojure in babashka

### DISCUSSION THREAD

**Discussion by borkdude** [babashka/babashka#917] [Show and tell]
I'm not the author, but I think it belongs in this forum, especially now that 4clojure is closing.

https://github.com/porkostomus/4bb

/cc @porkostomus

**Comment by bobbicodes** [comment]
I was thinking, since someone else is taking on a new web app, that the thing I built was intended for a particular use case, i.e. those times when you don't *want* to start a web app. Perhaps it would make sense for it to remain that way. As a recent historical parallel, there were the [Clojure Koans](http://clojurekoans.com/), and also the [Clojurescript Koans](http://clojurescriptkoans.com/).

That said, adding new features could be fun! Since we're working in a very similar domain with the new version of Exercism, I have this vision of cross-pollinating ideas from over there. In particular, the concept of building a *super-linter* (that's just what I'm calling it) to provide instant "canonical" feedback on common solution patterns seems like it could have valuable potential.

As it sits, the project is a glorified test-runner. Perhaps it could be elevated to the status of "learning platform"?

**Comment by bobbicodes** [comment]
Oh here's an idea. Lets say hypothetically that 4clojure is propaganda to indoctrinate folks into TDD.

Truth be told, I personally learned about unit testing by accident when I realized that's what 4clojure problems were. That was cool, but I recall my immediate reaction was that it seemed a slightly back-asswards way to teach how to *write functions*, so much so that I made a [whole repo](https://github.com/porkostomus/4clojure-unkoaned) where I just took the solutions and presented them as normal functions. Speaking of solutions, let's not forget about [coal mine](https://github.com/mfikes/coal-mine).

What do we do about it? Become the cult of generative testing. Replace the unit tests with specs. I'll put them [over here](https://github.com/porkostomus/4clojure-spec)
